CMakeLists.txt

```diff
@@ -253,23 +252,55 @@ if (CLR_CMAKE_PLATFORM_UNIX)
     message("Detected FreeBSD amd64")
   endif(CLR_CMAKE_PLATFORM_FREEBSD)
 
+  # set the CLANG sanitizer flags for debug build
+  if(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
+    # obtain settings from running enablesanitizers.sh
+    string(FIND "$ENV{DEBUG_SANITIZERS}" "asan" __ASAN_POS)
+    string(FIND "$ENV{DEBUG_SANITIZERS}" "ubsan" __UBSAN_POS)
+    if ((${__ASAN_POS} GREATER -1) OR (${__UBSAN_POS} GREATER -1))
+      # set compiler flags
+      # -fdata-sections -ffunction-sections: each function has own section instead of one per .o file (needed for -Wl and --gc-sections)
+      # -O1: basic level of optimization
+      # -fPIC: required for shared libraries (Position Independent Code)
+      # -fno-omit-frame-pointer: better stack dumps for debugging
+      set(CLR_SANITIZE_CXX_FLAGS "${CLR_SANITIZE_CXX_FLAGS} -fdata-sections -ffunction-sections -O1 -fPIC -fno-omit-frame-pointer -fsanitize=")
```

Comment:
Are you sure we need to set the -fPIC option globally? We set it in the specific CMakeLists.txt for libraries where it is needed, but not everywhere. 
Also, the -fno-omit-frame-pointer is already set further in this CMakeLists.txt (both for release and debug builds), but it is true it is not set for the build of stuff that's not using PAL, inclusing PAL itself (all the stuff around line 292 below), so it might be good to move that setting up and keeping the option for both release and debug builds.
Also, it is not clear to me why would the -fPIC /  -fno-omit-frame-pointer be in the CLR_SANITIZE_CXX_FLAGS when they are not specific to the sanitize.